### ISSUE TITLE
When the client updates, the updater, should look at the "general-config" for the correct jar-path

### ISSUE THREAD

**Issue by Antmanofgod** [Osiris-Team/AutoPlug-Client#111]
Importance: 7

Problem:
The general-config asks you where your server jar is, but when the Autoplug-Client.jar updates, and a restart occurs, the automatic updater, should reference the "general-config" where you server jar is, but it doesn't.
This causes the server to crash, and forces users to have to manually update their jar file.

Solution:
Since the location of the jar file has to be established anyway for the server to run properly, the general config should also serve the purpose of telling the updater where the server jar is, so it can find it and replace it and prevent a crash on restart.
![146616484-b20638b5-2f76-4a74-83b0-1ff6dd9dc7eb](https://user-images.githubusercontent.com/28590639/146650128-7c8764da-917a-48be-834b-55eae2a3708e.PNG)

**Comment by Osiris-Team** [comment]
@Antmanofgod Can u send your directory structure.

**Comment by Antmanofgod** [comment]
[Multicraft - Minecraft FTP Client.pdf](https://github.com/Osiris-Team/AutoPlug-Client/files/7835254/Multicraft.-.Minecraft.FTP.Client.pdf)

The "jar" folder is where the server jar is stored.

**Comment by Osiris-Team** [comment]
Ok but that means that the AutoPlug jar is inside the /jar folder but has its working directory still in the server root, where also your paper-latest.jar is located.
The `jar-path: auto-find` searches that directory and thus doesn't give us information about the location of the AutoPlug-Client.jar

**Comment by Antmanofgod** [comment]
Yeah cause I guess that's where the error lies, it goes looking for the client, but can't find it cause it doesn't know to look in the jar folder.

**Comment by Osiris-Team** [comment]
I tried implementing this feature until I realized that it would take a lot more work than expected.
The current self-updater is already fragile and complex, making it more complex than it already is wouldn't be good.
The workaround for your issue is also fairly simple namely: Disabling the self-updater and installing updates of autoplug manually.
Thus I decided to not implement this feature, sry.
